### ISSUE TITLE
error : Not a valid marker at offset 684, found: 140    <-- what??

### ISSUE THREAD

**Issue by pouu69** [danialfarid/ng-file-upload#1113]
upload image situation  i see this error =>  Not a valid marker at offset 684, found: 140  

what is it??

**Comment by danialfarid** [comment]
That's just a warning for an error in exif info of the file. That's shouldn't cause any problem. Only if the file has orientation exif info it means that it cannot rotate the file to fix the orientation

**Comment by danialfarid** [comment]
Leave this open I will try to make it a warning or suppress this

**Comment by voltechs** [comment]
This error is preventing multiple file uploads, when I drag a group of photos onto the drop zone, just FYI

**Comment by danialfarid** [comment]
@voltechs that should never happen, could you provide your sample file that causes this issue ( steps to reproduce )

**Comment by danialfarid** [comment]
Fixed at version 10.0.0

By default ngf-fix-orientation is off so you need to enable it. Also that error should not prevent upload anymore.

**Comment by Johnnyrook777** [comment]
Hi Daniel. When uploading files without EXIF data and setting fix rotate to true you get this error (v10.0.2). Would it be better to return null when checking the EXIF data rather than throwing an error?

**Comment by danialfarid** [comment]
What error do you see? @Johnnyrook777

**Comment by Johnnyrook777** [comment]
Two examples:
Not a valid marker at offset 623, found: 243
Not a valid marker at offset 623, found: 236

Seem you get this error if the EXIF data isn't present, in which case you should simply not rotate the image.

The orientation method currently rejects the promise if a string is found via the EXIF, I'm thinking it might be better to resolve a null. It may actually be better to check if the result is a number. Something like the code below

Sorry, I'm not super familiar with the library, so I'm not sure what the implications may be for this change

```javascript
upload.orientation = function (file) {
        if (file.$ngfOrientation != null) {
            return upload.emptyPromise(file.$ngfOrientation);
        }
        var defer = $q.defer();
        var fileReader = new FileReader();
        fileReader.onload = function (e) {
            var orientation;
            try {
                orientation = findEXIFinJPEG(e.target.result);
            } catch (e) {
                defer.reject(e);
                return;
            }

            // Proposed solution 1 - Check for a number result
            if (angular.isNumber(orientation)) {
                file.$ngfOrientation = orientation;
                defer.resolve(orientation);
            }
            else {
                 defer.resolve();
            }

            // Proposed solution 2
            if (angular.isString(orientation)) {
                // Johnnyrook Fix, 
                defer.resolve();
                // Original
                //defer.reject(orientation);
            } else {
                file.$ngfOrientation = orientation;
                defer.resolve(orientation);
            }
        };
        fileReader.onerror = function (e) {
            defer.reject(e);
        };

        fileReader.readAsArrayBuffer(file);
        return defer.promise;
    };
```

**Comment by danialfarid** [comment]
Can you attach the images that produce that error here?

**Comment by Johnnyrook777** [comment]
![l2e5cvl](https://cloud.githubusercontent.com/assets/10512977/11202300/c9da679a-8d3a-11e5-85a0-eca82fd78256.jpg)

**Comment by Johnnyrook777** [comment]
Pretty sure it will occur with any image that doens't have the EXIF properites.

Also, for reference. This is my button HTML

`<button class="btn btn-primary btn-file" ngf-select="uploadFiles($file)" ngf-resize="{width: 1024, height: 1024, quality: 0.95}" ngf-fix-orientation="true" accept="image/*" title="Add or change Photo">
                        Photo
                    </button>`

**Comment by danialfarid** [comment]
Alright will be fixed in the next version. For now you can just ignore the error since it doesn't affect the functionality.

**Comment by Johnnyrook777** [comment]
Thanks @danialfarid.

Yeah, I've got a quick workaround in our production version.

Thanks mate. Much appreciated

**Comment by carlosfunk** [comment]
It's been a while since I laughed reading some github comments, thanks for that picture @Johnnyrook777

**Comment by Johnnyrook777** [comment]
Hahaha, thanks @carlosfunk 

I actually found the issue when I uploaded that pic to our testing system .... Its now my goto picture for all  testing and/or profile pictures ;)

**Comment by danialfarid** [comment]
Fixed at version 10.0.3
